server/client_manager.py

```diff
@@ -344,6 +362,44 @@ def change_character(self, char_id, force=False):
 
             self.area.update_timers(self, running_only=True)
 
+        '''
+        def save_yt_cache(self):
+            with open("storage/yt_cache.yaml", "w", encoding="utf-8") as stream:
+                yaml.dump(self.yt_cache, stream, default_flow_style=False)
+
+        def load_yt_cache(self):
+            pass
+        '''
+
+        def mirror_youtube(self, yt_url):
+            try:
+                with yt_dlp.YoutubeDL(self.ydl_opts) as ydl:
+                    info = ydl.extract_info(yt_url, download=True)
+            except yt_dlp.utils.DownloadError:
+                raise ClientError("Server is not configured with ffmpeg. Please inform the staff of the server about the issue.")
+                print("YouTube support non-functional without ffmpeg. Please enable 'get_ffmpeg' in the config.yaml or install ffmpeg on your system and add it to the PATH environmental variable.") #FIXME: use logger
+
+
+            yt_song_path = info.get("requested_downloads")[0].get("filepath")
+            yt_file = open(yt_song_path, 'rb')
+            yt_config = self.server.config["youtube_play"]
+
+            r = requests.post(yt_config["request_url"],
+                              data=yt_config["args"],
+                              files={yt_config["file_form_name"]:yt_file}).content.decode("utf-8")
+            cache_entry = {
+                            "upload_time": datetime.datetime.now(),
+                            "song_url": r
+                    }
+            yt_file.close()
+            self.yt_cache[info["id"]] = cache_entry
+            try:
+                os.remove(yt_song_path)
+            except PermissionError:
+                pass #FIXME: unlucky
```

Comment:
For the best, I left this in while testing since I could not reproduce this bug on Linux, and was only happening on Windows (likely due to NOT closing the file stream. It should never occur now, but I will set up a virtual machine for further testing).